scenes/player/player_controller.gd

```diff
@@ -260,6 +194,9 @@ func Player_shot(player_damange: float)->void:
 	attack_impulse.apply_central_impulse(mouse_position * 1000)
 	get_parent().add_child(attack_impulse)
 
+func _on_dead()->void:
+	queue_free() 
+
```

Comment:
It breaks a lot . I found 2 different lines of code where the game breaks due to queuing free.

Reply:
But I I am not sure If it's problem of these lines of code but not of their lines of code